libraries/AP_Baro/AP_Baro_Backend.h

```diff
@@ -78,4 +78,18 @@ class AP_Baro_Backend
     void set_bus_id(uint8_t instance, uint32_t id) {
         _frontend.sensors[instance].bus_id.set(int32_t(id));
     }
+
+    bool _backend_exists(const uint32_t backend) {
```

Comment:
I think we need to restrict this check to I2C. The method here will cause issues for CAN devices.
Have a look at _have_i2c_driver() in AP_Compass and see if we can use a similar method here:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Compass/AP_Compass.cpp#L1023

Reply:
Done.

Thank you for guiding me.

Reply:
@tridge kindly advise.